hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -331,6 +332,35 @@ public synchronized X509Certificate getCACertificate() {
     return firstCertificateFrom(caCertPath);
   }
 
+  /**
+   * Return all certificates in this component's trust chain,
+   * the last one is the root CA certificate.
+   */
+  public synchronized List<X509Certificate> getTrustChain() {
+    CertPath path = getCertPath();
+    if (path == null || path.getCertificates() == null) {
+      return null;
+    }
+
+    List<X509Certificate> chain = new ArrayList<>();
+    // certificate bundle case
+    if (path.getCertificates().size() > 1) {
+      for (int i = 0; i < path.getCertificates().size(); i++) {
+        chain.add((X509Certificate) path.getCertificates().get(i));
+      }
+    } else {
+      // case before certificate bundle is supported
+      chain.add(getCertificate());
+      chain.add(getCACertificate());
```

Comment:
One thing I am not sure here...
Is it possible that we do not have a subordinate CA cert stored due to upgrading from a very old version or on a cluster without SCM HA? If so we might not want to add a null value to this list... however if the null value is possible, but does not cause any trouble, then let's just keep this piece simple.

Reply:
Although I think there is always CA certificate in the old version Ozone,  but it's safer to check the CA certificate before add it to the list.